### ISSUE TITLE
Unknown error when creating new task in a context

### ISSUE THREAD

**Issue by habernal** [TracksApp/tracks#2385]
Tracks 2.4.2

Repro:
* Go to any existing context
* Create a new task (just fill in the task name, that's enough; the context is pre-defined), click "Add action"
* A red message is shown "There was an error retrieving from server: error"
* The task is created but **not displayed** immediately, one has to refresh the site
* No apparent errors shown in JS Console or in the logs

**Comment by mattr-** [comment]
I was able to reproduce this. The error message is `undefined method 'context_id' for nil:NilClass`

Application backtrace:
```
app/controllers/todos_controller.rb:943 block (2 levels) in determine_down_count
lib/tracks/source_view.rb:17 context
app/controllers/todos_controller.rb:942 block in determine_down_count
lib/tracks/source_view.rb:43 source_view
app/controllers/todos_controller.rb:938 determine_down_count
app/controllers/todos_controller.rb:137 block (2 levels) in create
app/controllers/todos_controller.rb:121 create
```

**Comment by ZeiP** [comment]
This was fixed in #2477.
